File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/commands/ChargeFailedCommand.java
package com.hedvig.paymentservice.domain.payments.commands;

import lombok.Value;
import org.axonframework.commandhandling.TargetAggregateIdentifier;

import java.util.UUID;

@Value
public class ChargeFailedCommand {
  @TargetAggregateIdentifier
  String memberId;

  UUID transactionId;
}

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/commands/ChargeFailedCommand.kt
package com.hedvig.paymentservice.domain.payments.commands

import java.util.UUID
import org.axonframework.commandhandling.TargetAggregateIdentifier

data class ChargeFailedCommand(
    @TargetAggregateIdentifier  
    val memberId: String,
    val transactionId: UUID
)
